fix: use environment instead config file

ja7ad opened the pull request:
## Description

> This PR replace env instead yaml configuration

## Related Issue

Fixes [# (issue)
](https://github.com/ezex-io/Repo-ezeX-Development-Project/issues/132)

ja7ad committed:
feat: config: Load config from env vars

This commit removes yaml config and loads from env vars.
Also removes config validation.

ja7ad committed:
chore: Update dependencies and go version

ja7ad committed:
refactor: server: Use gopkg logger and address

This commit refactors the http and fasthttp servers to use the
`gopkg/logger` interface instead of `log/slog`. It also consolidates
address configuration.

ja7ad committed:
refactor(http_test): Simplify server initialization

- Removes server config struct
- Uses a simpler proxy rules map
- Simplifies server start/stop test

ja7ad committed:
feat: cmd/proxier: Load config from env and use gopkg

ja7ad committed:
docs: Update README with env var config

Refactors configuration instructions to use environment variables.
Includes examples and descriptions for each variable.
Removes YAML configuration section.

ja7ad committed:
feat: server: Implement fastHTTP server

ja7ad committed:
feat: Add example env file for proxier

b00f commented on .env.example:
```diff
@@ -0,0 +1,3 @@
+EZEX_PROXIER_ADDRESS=127.0.0.1:8081
+EZEX_PROXIER_ENABLE_FASTHTTP=false
+EZEX_PROXIER_PROXY_RULES=/foo|https://httpbin.org/get,/bar|https://google.com
```
In Rules we don't need to mention the HTTP Method? Like Get, or Post, etc?

ja7ad replied:
Not need

b00f commented on .env.example:
```diff
@@ -0,0 +1,3 @@
+EZEX_PROXIER_ADDRESS=127.0.0.1:8081
```
Rename it to `PROXIER_ADDRESS` 
It can be used by non-ezex projects

ja7ad committed:
fix: support JSON format for proxy rules

This commit introduces support for JSON format in proxy rules.
It enhances readability and maintainability.

ja7ad committed:
refactor: Rename env vars and update rules format

The environment variables were renamed for brevity.
Proxy rules are now parsed as JSON.
